spanner/spanner_samples.rb

```diff
@@ -1495,6 +1505,211 @@ def write_read_float64_array project_id:, instance_id:, database_id:
+  # [START spanner_create_backup]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+  # backup_id = "Your Spanner backup ID"
+  # expire_time = "Your Spanner backup expiration time"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+  database = instance.database database_id
+
+  job = database.create_backup backup_id, expire_time
+
+  puts "Backup operation in progress"
+
+  job.wait_until_done!
+
+  backup = instance.backup backup_id
+  puts "Backup #{backup.backup_id} of size #{backup.size_in_bytes} bytes was created at #{backup.create_time}"
+  # [END spanner_create_backup]
+end
+
+def restore_backup project_id:, instance_id:, database_id:, backup_id:
+  # [START spanner_restore_backup]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID of where to restore"
+  # backup_id = "Your Spanner backup ID"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+
+  backup = instance.backup backup_id
+  job = backup.restore database_id
+
+  puts "Waiting for restore backup operation to complete"
+
+  job.wait_until_done!
+
+  restore_info = job.database.restore_info
+  puts "Database #{restore_info.backup_info.source_database_id} restored from backup #{restore_info.backup_info.backup_id}"
+  # [END spanner_restore_backup]
+end
+
+def create_backup_cancel project_id:, instance_id:, database_id:, backup_id:, expire_time:
+  # [START spanner_cancel_backup_create]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+  # backup_id = "Your Spanner backup ID"
+  # expire_time = "Your Spanner backup expiration time"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+  database = instance.database database_id
+
+  job = database.create_backup backup_id, expire_time
+
+  puts "Backup operation in progress"
+
+  job.cancel
+  job.wait_until_done!
+
+  backup = instance.backup backup_id
+
+  backup.delete if backup
+
+  puts "#{backup_id} creation job cancelled"
+  # [END spanner_cancel_backup_create]
+end
+
+def list_backup_operations project_id:, instance_id:, database_id:
+  # [START spanner_list_backup_operations]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+
+  jobs = instance.backup_operations filter: "metadata.database:#{database_id} AND metadata.@type:CreateBackupMetadata"
```

Comment:
In other languages the metadata type filter needed to be metadata.@type:type.googleapis.com/google.spanner.admin.database.v1.CreateBackupMetadata. Is Ruby different?

Reply:
done